### PR TITLE
[quant][graphmode] Add tests for script models got from tracing and fix issues

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30210 [quant][graphmode] Add tests for script models got from tracing and fix issues**

Summary:
Changes to make quantizing traced model work:

0. aten::_convolution(landed by Lingyi)
1. Added ConstantPooling for traced graph to merge redundant constant nodes
2. Added FuseLinear to fuse exploded linear function into `aten::linear`
3. Refactored the code to identify weight and bias
4. Added weight/bias identification for aten::linear

Test Plan:
python test/test_quantization.py

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

### CODE REVIEW

**Review by ZolotukhinM — CHANGES_REQUESTED** [review]
This PR has too many independent changes piled in and is hard to review because of that. Please split out those changes into individual PRs (e.g.: 1) renaming stuff in tests without changing logic and formatting stuff, 2) factoring out isAtenFuncNthArg, 3) invoking more passes in insertObservers, 4)... )

File: torch/csrc/jit/passes/quantization.cpp [hunk 1]
```diff
@@ -588,8 +620,8 @@ void QuantizeHelper::quantizeTensors() {
     dequant = insertQuantDeQuantCall(v, qparams, scalar_type);
     v->replaceAllUsesWith(dequant->output());
     Node* q = dequant->input(0)->node();
-    // replaceAllUsesWith rewrote all uses of V, but we want to keep one: the one
-    // used in quant node. Restore it here:
+    // replaceAllUsesWith rewrote all uses of V, but we want to keep one: the
+    // one used in quant node. Restore it here:
```
Comment (lly-zero-one) [inline torch/csrc/jit/passes/quantization.cpp]: why do we get this back? I remember it is already properly handled in the insertQuantDequantCall.

Comment (jerryzh168) [reply torch/csrc/jit/passes/quantization.cpp]: maybe I'll need to rebase

**Comment by jerryzh168** [comment]
deprecated by stack: https://github.com/pytorch/pytorch/pull/30473

**Pull request closed without merging**